### DISCUSSION TITLE
`context!` macro

### DISCUSSION THREAD

**Discussion by JosephTLyons** [rwf2/Rocket#2001] [Questions]
Is this macro still valid?  I noticed it in this example:

https://github.com/SergioBenitez/Rocket/blob/v0.5-rc/examples/templating/src/tera.rs

But I'm not able to use it in the most recent releases:

```
rocket = "0.5.0-rc.1"
rocket_dyn_templates = { version = "0.1.0-rc.1", features = ["tera"] }
```

I get an error if I try to import it

`module context is private`

**Comment by definitelydiane** [comment]
See also [#1826](https://github.com/SergioBenitez/Rocket/issues/1826)

I copied the macro into my own code since the `context!` macro is not in a release version.
You can also create a context struct of your own; the macro just creates a struct for you as a convenience.

**Reply by JosephTLyons** [reply]
> See also [#1826](https://github.com/SergioBenitez/Rocket/issues/1826)
> 
> 
> 
> I copied the macro into my own code since the `context!` macro is not in a release version.
> 
> You can also create a context struct of your own; the macro just creates a struct for you as a convenience.

Right, this is what I'm currently doing with my rocket.rs web app - creating a context struct for each request handler.  There's definitively nothing wrong with it, but having the macro would clean up a lot of of those where the a context struct us only meant for a single request handler.
